site/en/blog/extension-instantnav/index.md

```diff
@@ -0,0 +1,179 @@
+and [“How do I determine when a page transitions?”](#transitions) sections.
+
+## How do I tell what life cycle the page is in? {: #lifecyle}
+
+The [documentLifecycle](/docs/extensions/reference/extensionTypes/#type-DocumentLifecycle)
+attribute has been added to a number of extensions APIs where the *frameId* was
+previously available. If the value is present on an event (such as
+[onCommitted](/docs/extensions/reference/webNavigation/#event-onCommitted)),
+the value is the state at which the event was generated. You can always query
+information from the [WebNavigation GetFrame/GetAllFrames](/docs/extensions/reference/webNavigation/#method-getFrame)
+APIs, but using the value from the event is always preferred. If you do use the
+[GetFrame](/docs/extensions/reference/webNavigation/#method-getFrame) API be
+aware the state of the frame may have changed from when the event was generated
+to when the GetFrame promise is resolved.
+
+The [documentLifecycle](/docs/extensions/reference/extensionTypes/#type-DocumentLifecycle)
+has the following values:
+
+- **prerender** : Not currently presented to the user but preparing to possibly be displayed to the user.
+- **active**: Presently displayed to the user.
+- **cached**: Stored in the back/forward cache.
+- **pending_deletion**:  A special case where the document is actively being destroyed.
+
+## How do I determine if a frame is the outermost frame? {: #outermost-frame}
+
+Previously extensions may have used checking [*frameId*](/docs/extensions/reference/webNavigation/#a-note-about-frame-ids)
+== 0 to determine if the event occurring is for the outermost frame or not.
+With multiple pages in a tab we now have multiple outermost frames, so the
+definition of *frameId* is problematic. You will never receive events about a
+Back/Forward cached frame. However, for prerender frames the *frameId* will be
+non-zero for the outermost frame. So using *frameId* == 0 as a signal for
+determining if it is the outermost frame is incorrect.
+
+To help with this, we introduced a new field called [*frameType*](/docs/extensions/reference/extensionTypes/#type-FrameType)
+so determining if the frame is indeed the outermost frame is now easy.
+
+The [*frameType*](/docs/extensions/reference/extensionTypes/#type-FrameType)
+has the following values:
+
+- **outermost_frame**: Typically referred to the topmost frame. Note that there are multiple of
+these: there could be many for the various documentLifecycles (such as a
+prerendered outermost_frame and a cached outermost_frame).
+- **fenced_frame**: A special type of frame that is embedded inside other frames
+(see [fenced-frames](https://github.com/WICG/fenced-frame))
+- **sub_frame**: Typically an iframe
+
+
+We can combine *documentLifecycle* with *frameType* and determine if a frame
+is the active outermost frame. ie.
+(documentLifecycle == “active” && frameType == “outermost_frame”)
```

Comment:
Prefer "for example" over latin abbreviations. Use monospace formatting for code snippets.

```suggestion
We can combine *documentLifecycle* with *frameType* and determine if a frame is
the active outermost frame. For example, `tab.documentLifecycle == "active"
&& tab.frameType == "outermost_frame"`.
```